src/main.cpp

```diff
@@ -1471,8 +1449,27 @@ bool AcceptToMemoryPool(CTxMemPool& pool, CValidationState &state, const CTransa
  
             // Bring the best block into scope
             view.GetBestBlock();
- 
-            nValueIn = view.GetValueIn(tx);
+
+            // If any of the inputs comes from a certificate, we make sure to have its sidechain in cache
+            // before rotating backend view to dummy
+            for(const CTxIn& in: tx.GetVin()) {
+                const CCoins *coins = view.AccessCoins(in.prevout.hash);
+                assert(coins);
+
+                if (coins->IsFromCert()) { //This is just to avoid following assert for non-cert coins
+                    // HaveInputs above checks for utxos availability. So accessing vout[in.prevout.n] is safe
+                    assert(coins->IsAvailable(in.prevout.n));
+
+
+                    //check on vout is better than coins->IsFromCert() since it'll skip loading scInfo for zero bwt amount certs
```

Comment:
This comment is not clear to me: how is it better?

Reply:
Probably poorly put, I am that we need to load in the cache system information about sidechains iif spent input is a certificate with backward transfers. Conversely, we won't load sidechains info for both ordinary tx and for certs with no bwt